tests/setup_test.go

```diff
@@ -7008,3 +7008,193 @@ func TestProjectCommitStatusRedelivery(t *testing.T) {
 		}
 	})
 }
+
+func TestRunRequiredEnvVariables(t *testing.T) {
+	t.Parallel()
+
+	config := `
+		{
+			runs: [
+			{
+				name: 'run01',
+				tasks: [
+				{
+					name: 'task01',
+					runtime: {
+					containers: [
+						{
+						image: 'alpine/git',
+						},
+					],
+					},
+					steps: [
+					  { type: 'run', command: 'env -u AGOLA_SSHPRIVKEY' },
+					  { type: 'run', command: 'echo %d' },
+					],
+				},
+				],
+			},
+			],
+		}
+	`
+
+	tests := []struct {
+		name      string
+		config    string
+		env       map[string]string
+		runNumber uint64
+	}{
+		{
+			name:      "test push with run count 1",
+			config:    fmt.Sprintf(config, 1),
+			runNumber: 1,
+			env: map[string]string{
+				"AGOLA_RUN_COUNTER":   "1",
+				"AGOLA_GIT_REF_TYPE":  "branch",
+				"AGOLA_GIT_REF":       "refs/heads/master",
+				"AGOLA_GIT_BRANCH":    "master",
+				"AGOLA_GIT_TAG":       "",
+				"AGOLA_GIT_COMMITSHA": "",
```

Comment:
From this looks like AGOLA_GIT_COMMITSHA should be empty. No need to define it here if you're going to populate it later from annotations.